src/fns/unconstrained_helpers.nr

```diff
@@ -259,13 +260,22 @@ pub(crate) unconstrained fn __barrett_reduction<let N: u32>(
     }
     let quotient_mul_modulus_u60: U60Repr<N, 4> = U60Repr::new(quotient_mul_modulus_normalized);
 
+    // convert the input into U60Repr
     let x_u60: U60Repr<N, 4> = U60Repr::new(x);
     let mut remainder_u60 = x_u60 - quotient_mul_modulus_u60;
-
+    // barrett reduction is quircky so might need to remove a few modulus_u60 from the remainder
     if (remainder_u60.gte(modulus_u60)) {
         remainder_u60 = remainder_u60 - modulus_u60;
         quotient_u60.increment();
     } else {}
+    if (remainder_u60.gte(modulus_u60)) {
+        remainder_u60 = remainder_u60 - modulus_u60;
+        quotient_u60.increment();
+    }
+    if (remainder_u60.gte(modulus_u60)) {
```

Comment:
This is kind of hacky but was a suggestion by Kesha. Because apparently, at least our implementation of `barrett_reduction` in barretenberg had an issue where the quotient might have been `+- 2` of the actual correct output. This made some values that were out of range to come back in range. Although it might be unnecessary with the fixes to `barrett_reduction`